packages/monitor-v2/src/bot-oo-v3/SettleAssertions.ts

```diff
@@ -0,0 +1,59 @@
+import { getContractInstanceWithProvider, Logger, MonitoringParams, OptimisticOracleV3Ethers } from "./common";
+import { logSettleAssertion } from "./BotLogger";
+import { paginatedEventQuery } from "../utils/EventUtils";
+import {
+  AssertionSettledEvent,
+  AssertionMadeEvent,
+} from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/OptimisticOracleV3";
+
+export async function settleAssertions(logger: typeof Logger, params: MonitoringParams): Promise<void> {
+  const oo = await getContractInstanceWithProvider<OptimisticOracleV3Ethers>("OptimisticOracleV3", params.provider);
+
+  const currentBlockNumber = await params.provider.getBlockNumber();
+
+  const assertions = await paginatedEventQuery<AssertionMadeEvent>(oo, oo.filters.AssertionMade(), {
```

Comment:
Thanks for adding this method!

Even after splitting up the query, trying to go all the way back to block zero will be really intensive. For instance polygon has a max query range of 3500 blocks.

Have you considered ways to limit how far back you search? For instance, we probably don't expect assertions to exist for more than a month without being settled, so we could set a configurable lookback of roughly a month for each chain.